pkg/asset/cluster/tfvars.go

```diff
@@ -400,15 +399,31 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		for i, w := range workers {
 			workerConfigs[i] = w.Spec.Template.Spec.ProviderSpec.Value.Object.(*machinev1beta1.GCPMachineProviderSpec)
 		}
+
+		publicZoneName := ""
+		publicZoneProject := ""
 		if installConfig.Config.Publish == types.ExternalPublishingStrategy {
-			publicZone, err := gcpconfig.GetPublicZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.BaseDomain)
-			if err != nil {
-				return errors.Wrapf(err, "failed to get GCP public zone")
+
+			if installConfig.Config.GCP.PublicManagedZone == "" {
+				publicZone, err := gcpconfig.GetPublicZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.BaseDomain)
+				if err != nil {
+					return errors.Wrapf(err, "failed to get GCP public zone")
+				}
+				publicZoneName = publicZone.Name
+			} else {
```

Comment:
Is this code taking care of the case where installConfig.Config.GCP.PublicManagedZone == "None"?